### PR TITLE
ensure that sequence blocks cannot be added or deleted after report is finalized

### LINKED ISSUE #6309: Finalizing CI Report does not lock all components
here is how a ci report presents itself after it has just been finalized.
image

the "add new" button and the "delete sequence block" buttons are still active, they should be gone/disabled.

image

current workaround: refresh the page.

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #6309

### CODE REVIEW

File: tests/acceptance/curriculum-inventory/leadership-test.js [hunk 1]
```diff
@@ -13,44 +13,48 @@ module('Acceptance | curriculum inventory leadership', function (hooks) {
...
     assert.equal(administrators.length, 2);
     assert.equal(administrators[0].text, '3 guy M. Mc3son');
     assert.equal(administrators[1].text, '4 guy M. Mc4son');
   });
 
   test('search administrators', async function (assert) {
-    assert.expect(11);
-    await page.visit({ reportId: 1, leadershipDetails: true });
-    await page.details.leadershipExpanded.manage();
+    const reportModel = await this.owner
```
Comment (jrjohnson) [inline tests/acceptance/curriculum-inventory/leadership-test.js:53]: I don't understand why you're pulling the report here only to use the id that you used to pull the report?

File: app/templates/curriculum-inventory-report/index.hbs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <CurriculumInventory::SequenceBlockList
   @report={{@model}}
   @sequenceBlocks={{this.topLevelSequenceBlocks}}
-  @canUpdate={{this.canUpdate}}
+  @canUpdate={{and (not @model.isFinalized) this.canUpdate}}
```
Comment (jrjohnson) [inline app/templates/curriculum-inventory-report/index.hbs:4]: One million lines of tests and this single change!! ❤️

File: tests/acceptance/curriculum-inventory/leadership-test.js (hunk 1 above)
Comment (stopfstedt) [reply tests/acceptance/curriculum-inventory/leadership-test.js:53]: yea that's wrong. fixing it...

Comment (stopfstedt) [reply tests/acceptance/curriculum-inventory/leadership-test.js]: handled.

Comment (jrjohnson) [reply tests/acceptance/curriculum-inventory/leadership-test.js]: Exists in a few other files as well. I guess I don't mind it, but it seems like the magic number is just being used to fetch a different thing from the DB whereas before it was being used directly. Having reportModel isn't a bad change though and is readable so I'll take it. Want me to mark the other files this is also needed in?

Comment (stopfstedt) [reply tests/acceptance/curriculum-inventory/leadership-test.js]: i found two more acceptance tests that use the magic report id. fixing them up now..

**Review by jrjohnson — APPROVED** [review]

**Review by dartajax — APPROVED** [review]
I tested this. Looks good. Gonna merge.

**Pull request merged**